Extension/src/LanguageServer/settings.ts

```diff
@@ -32,35 +47,56 @@ export class CppSettings extends Settings {
     }
 
     public get clangFormatPath(): string {
-        let path: string = super.Section.get<string>("clang_format_path");
+        let path: string = super.getWithFallback<string>("clangFormat.path", "clang_format_path");
         if (!path) {
             path = which.sync('clang-format', {nothrow: true});
         }
         return path;
     }
 
-    public get clangFormatStyle(): string { return super.Section.get<string>("clang_format_style"); }
-    public get clangFormatFallbackStyle(): string { return super.Section.get<string>("clang_format_fallbackStyle"); }
-    public get clangFormatSortIncludes(): string { return super.Section.get<string>("clang_format_sortIncludes"); }
-    public get clangFormatOnSave(): string { return super.Section.get<string>("clang_format_formatOnSave"); }
-    public get formatting(): string { return super.Section.get<string>("formatting"); }
+    public get clangFormatStyle(): string { return super.getWithFallback<string>("clangFormat.style", "clang_format_style"); }
+    public get clangFormatFallbackStyle(): string { return super.getWithFallback<string>("clangFormat.fallbackStyle", "clang_format_fallbackStyle"); }
+    public get clangFormatSortIncludes(): string { return super.getWithFallback<string>("clangFormat.sortIncludes", "clang_format_sortIncludes"); }
+    public get formatting(): string
+    {
+        let result: boolean;
+        let info = this.Section.inspect<boolean>("formatting.enabled");
+        if (info.workspaceFolderValue) {
+            result = info.workspaceFolderValue;
+        } else if (info.workspaceValue) {
+            result = info.workspaceValue;
+        } else if (info.globalValue) {
+            result = info.globalValue;
+        } else {
+            let value: string = this.Section.get<string>("formatting");
+            if (value === "Default" || value === "Disabled") {
+                return value;
+            }
```

Comment:
I think you need an `else` here that sets `result` to the default value or else we'll end up disabling formatting on line 76.

Or can you just call `getWithFallback` and translate the boolean result to Default | Disabled?  It seems like this function duplicates a lot of that code.  IOW, would this work?
```ts
    public get formatting(): string {
        return this.getWithFallback<boolean>("formatting.enabled", "formatting") ? "Default" : "Disabled";
    }
```

Reply:
Fixed.  As discussed, I'm accounting for settings under "formatting" potentially colliding with the deprecated "formatting" element.